homeassistant/components/spotify/media_player.py

```diff
@@ -449,6 +466,13 @@ def set_shuffle(self, shuffle: bool) -> None:
         """Enable/Disable shuffle mode."""
         self._spotify.shuffle(shuffle)
 
+    @spotify_exception_handler
+    def set_repeat(self, repeat: str) -> None:
+        """Set repeat mode."""
+        for spotify, home_assistant in REPEAT_MODE_MAPPING.items():
```

Comment:
It would be more efficient to create a reverse mode mapping constant to look up.